### ISSUE TITLE
Removing SklearnModel

### ISSUE THREAD

**Issue by rbharath** [deepchem/deepchem#1760]
At present the `SklearnModel` is a simple wrapper around Scikit-learn's API ([file](https://github.com/deepchem/deepchem/blob/master/deepchem/models/sklearn_models/__init__.py)). We added this wrapper in to make scikit-learn models fit into DeepChem's standard API so we could invoke models the same way regardless of whether they used Scikit-Learn or DeepChem's custom modeling capabilities.

After doing some thought, I'm starting to think we got this original design backwards. Scikit-learn's community is much larger than DeepChem's. It makes more sense that we defer to Scikit-learn's established API when using scikit-learn models. This would mean that users of DeepChem wouldn't have to learn a custom set of classes for using scikit-learn with MoleculeNet classes. 

I'm proposing we make the following changes:

1. Remove `SklearnModel` from DeepChem. This would entail deprecating `SklearnModel` in DeepChem 2.4 and removing it outright from DeepChem 2.5.
2. Swap over all examples and tutorials to use scikit-learn directly instead of using `SklearnModel`. This will require a bit of rewriting of some examples, but should be a limited amount of work and will reduce the cognitive overhead for our users. 

Does this plan make sense to folks? I'd love thoughts and feedback before we make any actual changes

CC @peastman @VIGS25

**Comment by peastman** [comment]
That sounds reasonable.

**Comment by rbharath** [comment]
Adding a comment here following up on discussion in #1916 

I've been doing a good amount of thinking and I'm less enthusiastic about removing `SklearnModel`. It's a thin wrapper that's straightforward to maintain and it's quite useful in practice. I've been making heavy use of `SklearnModel` recently in a few projects, and the nice thing is that it gives me apples-to-apples metric evaluation with GraphConvModel or other models.

I propose that we instead clean up `SklearnModel` to have clear documentation and just maintain it as a part of the library.

**Comment by rbharath** [comment]
CC @nd-02110114 @ncfrey who had some good comments in #1916
